plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -803,20 +832,15 @@ private static ResultSet getResultSet(PhoenixInputSplit split, QueryPlan queryPl
                 // For MR, skip the region boundary check exception if we encounter a split. ref: PHOENIX-2599
                 scan.setAttribute(SKIP_REGION_BOUNDARY_CHECK, Bytes.toBytes(true));
 
+                // TODO: How to make use of this with the new change.
                 ScanMetricsHolder scanMetricsHolder = ScanMetricsHolder.getInstance(
                         context.getReadMetricsQueue(),
                         physicalTableName.getString(),
                         scan,
                         phoenixConnection.getLogLevel());
 
-                TableResultIterator tableResultIterator = new TableResultIterator(
-                        phoenixConnection.getMutationState(),
-                        scan,
-                        scanMetricsHolder,
-                        services.getRenewLeaseThresholdMilliSeconds(),
-                        queryPlan,
-                        MapReduceParallelScanGrouper.getInstance());
-                iterators.add(LookAheadResultIterator.wrap(tableResultIterator));
+                ResultIterator resultIterator= queryPlan.iterator(MapReduceParallelScanGrouper.getInstance(), scan);
```

Comment:
Unrelated change? does not seem to be required for this change.
`scanMetricsHolder` is indeed no longer used.

Reply:
queryplan.iterator() change made it possible to get the count per split. With that change I wasn't sure how to make use of scanMetricsHolder. Can I remove that all together?